client/components/editor-standalone/utils.ts

```diff
@@ -24,3 +25,42 @@ export function omitFields<T extends IBaseRestApiResponse>(
 export function isMultiLineField(fieldSchema: IProfileSchemaType) {
     return (fieldSchema as IProfileSchemaTypeString)?.field_type === 'multi_line';
 }
+
+const FIELDS_TO_OMIT_FOR_COMPARISON = ['_links', '_planning_schedule', '_updates_schedule', 'event'];
+
+export function areEmbeddedItemsDirty<T extends Partial<IEventOrPlanningItem>>(original: T, diff: T) {
+    if (diff.type === 'event' && original.type === 'event') { // check both so type narrowing works
+        const origPlans = (original.associated_plannings ?? []);
+        const diffPlans = (diff.associated_plannings ?? []);
+
+        if (origPlans.length !== diffPlans.length) {
+            return true;
+        }
+
+        const plansOriginal = origPlans.map((x) => {
+            const itemCleaned = omit(x, FIELDS_TO_OMIT_FOR_COMPARISON);
+
+            if (isMoment(itemCleaned.planning_date)) {
+                itemCleaned.planning_date = itemCleaned.planning_date.toISOString();
+            }
+
+            return itemCleaned;
+        }).sort((a, b) => a._id.localeCompare(b._id));
+        const plansDiff = diffPlans.map((x) => {
+            const itemCleaned = omit(x, FIELDS_TO_OMIT_FOR_COMPARISON);
+
+            if (isMoment(itemCleaned.planning_date)) {
+                itemCleaned.planning_date = itemCleaned.planning_date.toISOString();
+            }
+
+            return itemCleaned;
+        }).sort((a, b) => a._id.localeCompare(b._id));
```

Comment:
This looks very similar. I'd suggest to extract the `map-clean-sort` logic into a separate function to avoid duplication. Something like this

```typescript
const cleanAndSortPlans = (plans: any[]) => {
    return plans
        .map((item) => {
            const cleanedItem = omit(item, FIELDS_TO_OMIT_FOR_COMPARISON);
            if (isMoment(cleanedItem.planning_date)) {
                cleanedItem.planning_date = cleanedItem.planning_date.toISOString();
            }
            return cleanedItem;
        })
        .sort((a, b) => a._id.localeCompare(b._id));
}
```

and then reuse it. What do you think?